client/my-sites/upgrades/domain-management/dns/txt-record.jsx

```diff
@@ -57,7 +64,12 @@ const TxtRecord = React.createClass( {
 						onChange={ this.props.onChange }
 						value={ this.props.fieldValues.data }
 						placeholder={ this.translate( 'e.g. %(example)s', { args: { example: 'v=spf1 include:example.com ~all' } } ) } />
-					{ ! isDataValid ? <FormInputValidation text={ this.translate( 'Invalid TXT Record' ) } isError={ true } /> : null }
+					{ hasNonAsciiData && (
+						<div className="form-input-validation dns__warning">
```

Comment:
Thought about it, kind of quit because it makes sense that the validation is boolean. It's a rather specific case.